lib/internal/source_map/source_map_cache.js

```diff
@@ -195,11 +190,7 @@ function rekeySourceMap(cjsModuleInstance, newInstance) {
 function sourceMapCacheToObject() {
   const obj = ObjectCreate(null);
 
-  const it = MapPrototypeEntries(esmSourceMapCache);
-  let entry;
-  while (!(entry = MapIteratorNext(it)).done) {
-    const k = entry.value[0];
-    const v = entry.value[1];
+  for (const [k, v] of esmSourceMapCache) {
```

Comment:
Does the destructuring here call an iterator (on the iterated value, i.e. on the map entry) that is modifiable through userland? I know that iterating the map is safe now, that that is also the case for the entries is not entirely obvious :)

Reply:
You're right, it does call `ArrayIteratorPrototype%.next` on the destructuring!

<details>
<summary>Repro code</summary>

```js
const { bind, call } = Function.prototype;
const uncurryThis = call.bind(bind, call);
function copyProps(src, dest) {
  for (const key of Reflect.ownKeys(src)) {
    if (!Reflect.getOwnPropertyDescriptor(dest, key)) {
      Reflect.defineProperty(
        dest,
        key,
        Reflect.getOwnPropertyDescriptor(src, key)
      );
    }
  }
}
const createSafeIterator = (factory, next) => {
  class SafeIterator {
    constructor(iterable) {
      this._iterator = factory(iterable);
    }
    next() {
      return next(this._iterator);
    }
    [Symbol.iterator]() {
        return this;
    }
  }
  Object.setPrototypeOf(SafeIterator.prototype, null);
  Object.freeze(SafeIterator.prototype);
  Object.freeze(SafeIterator);
  return SafeIterator;
};
function makeSafe(unsafe, safe) {
  copyProps(unsafe.prototype, safe.prototype);
  copyProps(unsafe, safe);
  if (Symbol.iterator in unsafe.prototype) {
    const createIterator = uncurryThis(unsafe.prototype[Symbol.iterator]);
    const next = uncurryThis(createIterator(new unsafe()).next);
    const SafeIterator = createSafeIterator(createIterator, next);
    safe.prototype[Symbol.iterator] = function () {
      return new SafeIterator(this);
    };
  }
  Object.setPrototypeOf(safe.prototype, null);
  Object.freeze(safe.prototype);
  Object.freeze(safe);
  return safe;
}

const SafeMap = makeSafe(Map, class SafeMap extends Map {
    constructor(i){super(i)}
});
const map = new SafeMap([['test', 'value']])

ArrayIterator.next = function next() {
    throw new Error('must not be called')
}

for(const [k,v] of map);
```

</details>

Reply:
```suggestion
  for (const entry of esmSourceMapCache) {
    const k = entry[0];
    const v = entry[1];
```

Reply:
Yeah, with that resolved I think this PR is good to go as far as I am concerned :+1: